xtask/src/main.rs

```diff
@@ -39,29 +54,12 @@ fn run_action(workspace: Workspace, target: &str, extra_args: &[String], feature
         // understand `+nightly`. So the easiest way to run `cargo fmt` for the nightly
         // toolchain is to use `rustup run nightly cargo fmt`.
         let mut cmd = Command::new("rustup");
-        cmd.args(&["run", "nightly", "cargo", target]);
-
-        let this_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-        let ws_dir = match workspace {
-            Workspace::Internal => [&this_dir, "..", "internal_ws"].iter().collect::<PathBuf>(),
-            Workspace::External => [&this_dir, ".."].iter().collect::<PathBuf>(),
-        };
-        cmd.current_dir(ws_dir);
-
+        cmd.args(&["run", "nightly", "cargo", target])
+            .current_dir(workspace.path());
         cmd
     } else {
         let mut cmd = Command::new(env::var("CARGO").unwrap());
-        cmd.arg(target);
-
-        let this_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
-        let ws_dir = match workspace {
-            Workspace::Internal => [&this_dir, "..", "internal_ws", "ykshim"]
```

Comment:
There should be no functional change. All of that logic just moved in to `Workspace::path()` since we had this code duplicated in two places. Or am I wrong?

Reply:
The new path is `&[&xtask_dir, "..", "untraced"]`, while before it was `&[&xtask_dir, "..", "untraced", "ykshim"]`. For fmt the first path is correct, but for building the second path is correct.

Reply:
I see. The paths are not the same after all.

Would we expect a crash though? Nothing is exploding right now. Could it be that by running in the crate root, it is recursing into `ykrt` anyway?

Reply:
If it doesn't give a build error, it is fine. I am just surprised that it works.